### DISCUSSION TITLE
How to debug "rook ceph operator" at local?

### DISCUSSION THREAD

**Discussion by zhucan** [rook/rook#10562] [Q&A] [answered]
https://github.com/rook/rook/blob/1ae867049b49079b76696e68ee9b8f30216528bd/cmd/rook/ceph/operator.go#L67

If we want to debug "rook ceph operator", why we need to check the pod is running?

**Comment by parth-gr** [comment]
For debugging Rook-ceph-operator you can turn on the debug logs in the operator.yaml and see the logs,

Otherwise at https://github.com/rook/rook/blob/1ae867049b49079b76696e68ee9b8f30216528bd/cmd/rook/ceph/operator.go#L67

We are quite getting the pod resources from this function and seeing if it matches the `ResourceRequirements` before proceeding ahead.

**Comment by zhucan** [comment]
we don't want to rebuild a new image, we want to run the "ceph operator" with a bin file and can use dlv to debug.

**Reply by parth-gr** [reply]
I haven't used dlv, but I would suggest that you can install it externally and start from https://github.com/rook/rook/blob/a730d6d8019816590207c108bf935af0c034885d/cmd/rook/main.go#L27

**Comment by travisn** [comment, marked as answer]
@zhucan A number of attempts have been made in the past for a debugging environment like you are trying, but the operator just has too many requirements for running in a full K8s environment. If you need to debug, the approach is to either look at the operator logs as parth mentioned, or to debug a unit test that exercises the code in question.

**Reply by zhucan** [reply]
ok, thanks
